src/CP/core/model.jl

```diff
@@ -364,17 +366,26 @@ end
 """
     global_domain_cardinality(model::CPModel)
 
-Return the number of variable-value edges in the tripartite graph 
-(or equivalently the sum of domain cardinalities of the variables), excluding the objective variable.
+Returns the sum of the cardinalities of the variable domains.
 """
 function global_domain_cardinality(model::CPModel)
     cardinality = 0
     for (id, x) in model.variables
-        if x != model.objective && !isa(x, IntVarView) && !isa(x, BoolVarView)
-            if isa(x.domain,BoolDomain)
-                cardinality += length(x.domain.inner.values)
-            else
-                cardinality += length(x.domain.values)
+        if isa(x.domain,BoolDomain)
+            cardinality += length(x.domain.inner.values)
+            if !isempty(x.children)
+                for child in x.children
+                    cardinality += length(child.domain.inner.values)
+                end
+            end
+        elseif isa(x.domain,IntSetDomain)
+            cardinality += length(x.domain)
+        else
+            cardinality += length(x.domain)
+            if !isempty(x.children)
+                for child in x.children
+                    cardinality += length(child.domain)
+                end

```

Comment:
We now take all variables, including `VarViews` into consideration in `global_domain_cardinality`